input-profile.ts

```diff
@@ -0,0 +1,322 @@
+import {Component, Object3D} from '@wonderlandengine/api';
+import {HandTracking} from '@wonderlandengine/components';
+import {vec3, quat} from 'gl-matrix';
+import {property} from '@wonderlandengine/api/decorators.js';
+
+const _TempVec = vec3.create();
+const _TempQuat = quat.create();
+const minTemp = new Float32Array(3);
+const maxTemp = new Float32Array(3);
+const hands = ['left', 'right'];
+
+interface VisualResponse {
+    target: Object3D;
+    min: Object3D;
+    max: Object3D;
+    id: string;
+}
+
+export class InputProfile extends Component {
+    static TypeName = 'input-profile';
+
+    private _gamepadObjects: Record<string, any> = {};
+    private controllerModel: any;
+    private toFilter: Set<string> = new Set(['vr-mode-active-mode-switch']);
+    private defaultControllerComponents?: Component[];
+    private handedness!: string;
+    private url!: string;
+    private ProfileJSON: any = null;
+    private modelLoaded!: boolean;
```

Comment:
Remove this variable, you already have a function to check if the model is loaded